draft-ietf-jsonpath-base.md

```diff
@@ -1502,8 +1502,9 @@ The well-typedness of function expressions can now be defined in terms of this t
 
 For a function expression to be well typed:
 
-1. its declared type must be well typed in the context in which it occurs, and
-2. its arguments must be well typed for the declared type of the corresponding parameters.
+1. its function name must be registered (see {{iana-fnex}}),
```

Comment:
```suggestion
1. its function name must be registered (see {{iana-fnex}}), which provides its declared type and the declared types of its parameters;
```

this would make it clear why that is a requirement.

(Do we want to spend MUST interoperability requirements here?)

Reply:
@cabo I made a similar change in 5f210fed5a2987e586f197d782cc1c4cf8edd824. Note that the bullets (1), (2), and (3) correspond to the summary points 1, 2, and 3.

Reply:
Section 2.1 already states:
> A JSONPath implementation MUST raise an error for any query which is not
well formed and valid.

So I don't think we need to use MIST here also.

Reply:
We don't need to, but we could.  I think it would help.

Reply:
Ok, see what you make of 3819d94bc7e821cb529003293dee54ec680cd43b.

Reply:
Thanks, that is what I meant.